Tools/peg_generator/scripts/benchmark.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3.9
+
+import argparse
+import ast
+import sys
+import os
+import resource
+from time import time
+
+import memory_profiler
+
+sys.path.insert(0, os.getcwd())
+from peg_extension import parse
+from pegen.build import build_parser_and_generator
+from scripts.test_parse_directory import parse_directory
+
+argparser = argparse.ArgumentParser(
+    prog="benchmark",
+    description="Reproduce the various pegen benchmarks"
+)
+argparser.add_argument(
+    "--parser",
+    action='store',
+    choices=['pegen', 'cpython'],
+    default='pegen',
+    help='Which parser to benchmark (default is pegen)'
+)
+argparser.add_argument(
+    "--target",
+    action="store",
+    choices=["xxl", "stdlib"],
+    default="xxl",
+    help="Which target to use for the benchmark (default is xxl.py)"
+)
+
+subcommands = argparser.add_subparsers(title="Benchmarks", dest="subcommand")
+command_compile = subcommands.add_parser("compile", help="Benchmark parsing and compiling to bytecode")
+command_parse = subcommands.add_parser("parse", help="Benchmark parsing and generating an ast.AST")
+command_check = subcommands.add_parser("check", help="Benchmark parsing and throwing the tree away")
+
+
+def benchmark(func):
+    def wrapper(*args):
+        times = list()
+        for _ in range(3):
+            start = time()
+            result = func(*args)
+            end = time()
+            times.append(end-start)
+        memory = memory_profiler.memory_usage((func, args))
+        print(f"{func.__name__}")
+        print(f"\tTime: {sum(times)/3:.3f} seconds on an average of 3 runs")
+        print(f"\tMemory: {max(memory)} MiB on an average of 3 runs")
+        return result
+    return wrapper
+
+
+@benchmark
+def time_compile(source, parser):
+    return (compile(source, os.path.join('data', 'xxl.py'), 'exec')
+            if parser == 'cpython'
+            else parse.parse_string(source, mode=2))
```

Comment:
IMO this would be clearer if you just used an `if`/`else` statement with two `return` clauses.